Proposed/asc056.md

```diff
@@ -0,0 +1,47 @@
+#### **Title**
+Add `informatics` to Autoprotocol instruction schema
+
+#### **Authorship**
+Asuka Ota <asuka.ota@strateos.com>
+
+#### **Motivation**
+In the current form, Autoprotocol is unable to capture a scientific intent at the molecular
+level upon execution of an `instruction`. One of the examples is pcr whereby a mixture of nucleotides
+and polymerase are mixed in a vial to synthesize new molecules via `thermocycle`. In another case, a user may want
+to purify a compound of interest from a mixture of compounds in instructions such as `spe` and/or `lcms`. In 
+addition to the physical execution parameters, there should be a set of parameters describing the intended effects 
+of an instruction on the sample composition.
+
+#### **Proposal**
+Instruction currently has `op`, and `data` attributes. We propose adding a new attribute called `informatics` 
+in our Autoprotocol schema to represent the intended effects of an instruction. A vendor may choose to integrate 
+the data specified in `informatics` with their internal LIMS systems. This `informatics` is an optional parameter 
+that can accept a list of effects for each instruction. `type` specifies the type of the effect, and `details` is a 
+set of parameters to describe the effect. We are proposing to only add 'attach_compounds' at this time, but more types 
+may be available in the future such as 'purify_compounds', 'liquefy', etc. Here, `refs` is a list of aliquots that are 
+affected by the instruction, and `compounds` is a list of compounds that will be attached to the `refs` (for 
```

Comment:
should `refs` here be limited to the set of refs involved in this Instruction?

Reply:
It should not, as it would not be of use to us for our initial use case. Recall, for chemical synthesis protocols, we want to empower the protocol author to be able indicate when (i.e. upon which instruction) to attach the specified final product to the final product container. Most likely, this will be upon an early `liquid_handle` or `provision` instruction, neither of which bear a `ref` to the final product container.

Reply:
hmm. For `attach_compounds` I think it would always be refs referred to in the instruction.

I suppose for something like `purify_compounds` compounds are often purified into the output aliquots instead of the input aliquot. e.g. 700uL of 1 mL input aliquot containing a crude mixture is injected in `lcms` to purify coi... At the end, the input aliquot would still have (i.e. 300uL of) the crude mixture, and the output aliquot(s) contain the purified coi. In those cases, we need to have a way to specify aliquots that do not exist yet in the system? Looks like `spe` has `destination_well` but not all 'purification' instructions have such params...

Anyhow, I agree with @dzleidig that for our current use case, this is safe?

Reply:
Yea.. that seems like an edge-case which we probably don't want to adopt out of the gate. 

If we're in agreement here, let's restrict this to our scenario and state that the refs here would have to be present in the instruction